src/lib/datepicker/datepicker.spec.ts

```diff
@@ -486,6 +487,81 @@ describe('MdDatepicker', () => {
         expect(cells[1].classList).not.toContain('mat-calendar-body-disabled');
       });
     });
+
+    describe('datepicker with change and input events', () => {
+      let fixture: ComponentFixture<DatepickerWithChangeAndInputEvents>;
+      let testComponent: DatepickerWithChangeAndInputEvents;
+      let inputEl: HTMLInputElement;;
+
+      beforeEach(async(() => {
+        fixture = TestBed.createComponent(DatepickerWithChangeAndInputEvents);
+        fixture.detectChanges();
+
+        testComponent = fixture.componentInstance;
+        inputEl = fixture.debugElement.query(By.css('input')).nativeElement;
+
+        spyOn(testComponent, 'onChange');
```

Comment:
Another option would be to use `jasmine.createSpy()` here. It would remove the need for creating the empty functions in lines 720 on.

```ts
testComponent.onChange = jasmine.createSpy('onChange');
testComponent.onInput = jasmine.createSpy('onInput');
...
```